Mono.Addins/Mono.Addins.Database/AddinScanner.cs

```diff
@@ -527,7 +527,29 @@ bool ScanDescription (IProgressStatus monitor, IAssemblyReflector reflector, Add
 					}
 				}
 			}
-			
+
+			// Fix up ModuleDescription so it adds assembly names.
+			foreach (ModuleDescription module in config.AllModules) { 
+				foreach (var s in module.Assemblies) { 
+					string asmFile = Path.Combine (config.BasePath, Util.NormalizePath (s));
+					object asm = reflector.LoadAssembly (asmFile);
+					module.AssemblyNames.Add (reflector.GetAssemblyFullName (asm));
```

Comment:
What about `AssemblyName.GetAssemblyName()`?

Reply:
Ah, it's serializable. That could work.

Reply:
I don't mean to use AssemblyName to store the name, I'm just proposing to use the GetAssemblyName method to get the name of the assembly from the file, instead of loading the assembly and then getting the name.

Reply:
Oh, that could work even better, thanks!